pkg1/elastic/elasticsearch.go

```diff
@@ -0,0 +1,190 @@
+package elastic
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"github.com/elastic/go-elasticsearch/v7"
+	"github.com/elastic/go-elasticsearch/v7/esapi"
+	"logexplorationapi/pkg1/constants"
+	"logexplorationapi/pkg1/logs"
+	"net/http"
+	"strings"
+)
+
+type ElasticRepository struct {
+	esClient *elasticsearch.Client
+}
+
+func InitializeElasticSearchClient() *elasticsearch.Client {
+
+	esAddress := flag.String("es-address", "https://localhost:9200", "ElasticSearch Server Address")
```

Comment:
It's a good practice to move these `flag` entries into something like `ApplicationConfig` structure with related fields, in the constructor you'd fill these fields with the values from `flag` entries. Then you could pass them as parameters into this `InitializeElasticsearchClient()` method. This would simplify adding more parameters as the configuration is done in one place. It also simplifies unit testing. Here is an example: https://github.com/Pronuntio/core/blob/fix/refactoring-packages/configuration/app.go#L5 and how it's used https://github.com/Pronuntio/core/blob/fix/refactoring-packages/cmd/server/main.go#L22.